### ISSUE TITLE
*CheckValidation* - insufficient access rights on object id-guest user

### ISSUE THREAD

**Issue by cadillacben** [alexed1/LightningFlowComponents#973]
### Steps to reproduce

1. Have a guest user to use and get the user Id
2. Modify the below code to run as your guest user and properly create an account without errors (mine is a person account)
3. Run the below test class (try not to laugh too much)
4. See error by filtering for "message"

### Expected behaviour
Test inserts should return zero errors.

### Actual behaviour
Some have no errors, but typically the last one has an error related to insufficient access.
  If commit=true I'll see something like 00177000002dMrFAAU [insufficient access rights on cross-reference id]
  If commit=false I'll see 00177000002dMrKAAU [invalid cross reference id]
  In actual practice in a flow I had it email me the error output and got this  [insufficient access rights on object id]

The actual flow has the action accept a single account record variable and commit=false.  I have never been able to recreate it as myself or as that guest user.  I think it's related to multiple people hitting the flow at the same time?  Some work fine, others don't, and eventually they get the record to insert.

The flow runs in system mode.

The error started being reported within the last week, but the flow and class have not changed.  I did check status.salesforce.com and didn't see any maintenance in the last 30 days.  I can only guess that the volume increased enough to cause this and SF is bulking the requests.

### Test class
```
@IsTest
public with sharing class testValidate {
    
    static testMethod void test() {
        
       User guestUser = [SELECT Id FROM User WHERE Id = '0050g000005oCPXAA2'];
       // run as guest user
       System.runAs (guestUser) {
            
       List<Account> acctInserts = new List<Account>();
		
		for(Integer i = 0; i < 5; i++)
        {
			acctInserts.add(
				new Account(FirstName='Ben',LastName='Account '+ String.valueOf(i),RecordTypeId = '012700000001dvoAAA',PersonMailingCountryCode = 'US',PersonMailingStateCode = 'nc',PersonMailingPostalCode = '27330')
			);
        }
				
        CheckValidation.Requests testRequest = new CheckValidation.Requests();
        List<CheckValidation.Requests> testRequestList = new List<CheckValidation.Requests>();
		List<CheckValidation.Results> testResultList = CheckValidation.checkForValidationErrors(testRequestList);
		List<String> allErrors = new List<String>();
		
// Test person inserts
for(Account a : acctInserts){
        System.debug('Test insert '+a.LastName);
        testRequest.inputCollection = null;
		a.FirstName = 'Ben'+system.now();
        testRequest.inputRecord = a;
        testRequest.isCommit = true;
        testRequest.checkDuplicates = false;
        //testRequestList.clear();
        testRequestList.add(testRequest);
	/*send outside the loop	
                testResultList = CheckValidation.checkForValidationErrors(testRequestList);
		
		system.debug('insert1 is error-'+testResultList[0].isError);
		system.debug('error message1 is-'+testResultList[0].errorMessages);
		allErrors.Add(testResultList[0].errorMessages);
	*/	
        system.debug('testRequestList is: '+testRequestList);
    }
    
	testResultList = CheckValidation.checkForValidationErrors(testRequestList);
	
    system.debug('insert1 is error-'+testResultList[0].isError);
	system.debug('error message1 is-'+testResultList[0].errorMessages);
    

    

    For(CheckValidation.Results r : testResultList)
        {
            allErrors.Add(r.errorMessages); 
            system.debug('insert is error loop-'+r.errorMessages);
        }
		
    }
    }
    }
```

</details>

**Comment by cadillacben** [comment]
I noticed that the debug statement inside the loop has no errors, but the one at the end does.  Not sure what that means.

**Comment by cadillacben** [comment]
[apex-07L77000000QvGSEA0.log](https://github.com/alexed1/LightningFlowComponents/files/7570459/apex-07L77000000QvGSEA0.log)

**Comment by cadillacben** [comment]
Here is a log of one that worked as expected.
[validationSuccess.log](https://github.com/alexed1/LightningFlowComponents/files/7571796/validationSuccess.log)

**Comment by ericrsmith35** [comment]
The action will try to "update" rather than "insert" a record if the record being processed has a value in the Id field.

**Comment by cadillacben** [comment]
That helps.  I noticed in the log that the account var had an ID after I assigned it's values, which was odd because these are inserts vs. updates.  I discovered that the mystery Id value was coming from them looping back around and running the flow again, which I enable so they can process multiple records.  Adding an assignment to clear the single record var seems to have resolved it.

Sorry for the fire drill, but thanks for the lead resolving it.
